[#12] RecommendFeedCard 컴포넌트 구현

ShinJongUng opened the pull request:
## Summary
- RecommendFeedCard와 그 밑의 SVG Icon 작업을 진행하였습니다. 
- (#17 )에서 구현된 PressableIcon과 PressableIcon 2개 (Upload, Like) 코드를 재사용 컴포넌트 위치인 utils로 옮기는 작업을 진행하였습니다.
- ios에서 RNSVG가 없던 오류를 pod install를 통해 해결하였습니다.
- RecommendFeedCard 안에 있는 icon의 위치를 잡아주었습니다. 

## Comments
- RecommendFeedCard.styles.tsx를 보시면 제가 Icon을 그냥 flexDirction과 dirction을 통해 layout을 잡아주었습니다. 디자이너님들이 figma를 통해 작업하시는 내용과는 다르게 진행하는 것이 좋겠습니다. (https://reactnative.dev/docs/flexbox) 공식문서를 참고 해주세요.
- 이렇게 위치를 잡아도 되는건지 아직 모르겠긴 한데, 공식 문서에서 추천하는 방향대로 작업 하였습니다.

ShinJongUng committed:
✨ feat: add RecommendFeedCard Component styles (#12)

RecommendFeedCard 컴포넌트의 스타일 추가

#12

ShinJongUng committed:
♻️ refactor: pressableIcon Component change directory src/components -> src/components/utils (#12)

PressableIcon 컴포넌트는 재사용 해야할 곳이 많아 utils 디렉토리로 옮김

#12

ShinJongUng committed:
♻️ refactor: pressable Icons change directory src/components -> src/components/utils (#12)

재사용 가능성이 높은 Icon 버튼들을 utils 폴더로 옮겨줌

#12

ShinJongUng committed:
🏗️ build: pod install RNSVG (#12)

ios ui manager를 위해 pod install 해주었습니다.

#12

ShinJongUng committed:
✨ feat: add RecommendFeedCard component (#12)

추천탭 안에 RecommendFeedCard 컴포넌트 작업

#12

vgihan commented on src/components/RecommendFeedCard/RecommendFeedCard.styles.tsx:
```diff
@@ -0,0 +1,19 @@
+  direction: 'rtl',
+  paddingVertical: 16,
+  paddingHorizontal: 15,
+});
```
paddingVertical은 widthPercentage 못먹이나요??

ShinJongUng replied:
아 맞네요 이것도 수정하겠습니다

vgihan commented on src/components/RecommendFeedCard/RecommendFeedCard.styles.tsx:
```diff
@@ -0,0 +1,19 @@
+  width: '100%',
+  height: '100%',
+  flexDirection: 'column-reverse',
+  direction: 'rtl',
```
안드로이드에서 테스트 해봤는데, direction rtl 이 안먹혀서 혹시 alignItems: 'flex-end' 하는 건 어떨까요 ?

ShinJongUng replied:
혹시 flex end가 먹히던가요?? 어제 해보니깐 안되던거 같아서 좀이따가 다시 확인해볼게요

vgihan replied:
flex-end 는 먹히는데, rtl이 먹히지 않았습니다 ㅜㅜ

ShinJongUng replied:
제가 안드로이드에서 구동 해봤는데, 따로 문제가 없었습니다. 아마도 당시 코드 작성하실때 width 값을 넣어주지않아서 발생한 문제로 보입니다. width 없이 alignSelf를 사용하면 되긴해서 일단 이걸로 고쳐서 publish 하겠습니다.

ShinJongUng replied:
코드 한번더 확인해주시면 제가 머지할게요~!

vgihan replied:
넵 지금 바로 확인하겠습니당

vgihan reviewed (approved):
Pressables 옮겨 주셔서 감사합니다 :) 일단 approve 할게용

ShinJongUng committed:
♻️ refactor: change styles (direction -> alignSelf) (#12)

안드로이드 상에 문제가 생길 수 있는 direction을 alignSelf로 변경하고, padding 값을 반응형으로 적용 해주었습니다.

#12

ShinJongUng committed:
🐛 fix: incorrect code of width,height percentage function (#12)

padding vertical에 width 함수를 작성하고 padding horizontal에 height를 작성했네요 수정합니다.

#12

vgihan commented on src/components/RecommendFeedCard/RecommendFeedCard.tsx:
```diff
@@ -0,0 +1,15 @@
+  );
+};
+
+export default RecommendFeedCard;
```
종웅님 늦게 말씀드려 죄송한데, 혹시 image는 어디 들어갈까요 ?

ShinJongUng replied:
이거 View로 작성해서.. children 값으로 넣으면 될거라고 생각했는데 따로 만들어줘야 할까요?

vgihan replied:
Image 컴포넌트를 line7과 line8 사이에 넣어야할 것 같은데, Image 컴포넌트도 style 수정이 필요할 것 같아서, props에서 image url 을 전달 받아서 Image 컴포넌트로 전달 해주는 방식이 괜찮을 것 같아요

ShinJongUng replied:
그렇게 작업 해야하는줄 몰랐네요 ㅎㅎ... 그 이건 이 pr에서 고치고 square card는 issue 새로 하나 파서 수정하겠습니다

ShinJongUng replied:
수정 완료했습니다. props로 받아줬는데 나중에 백엔드랑 연결할때 다시 손봐야할거 같습니다. fastimage 사용해서 resizeMode 적용시켜놨습니다.

vgihan replied:
백엔드랑 연결하는 로직은 상위 컴포넌트에서 수행하면 돼서 FeedCard는 그대로 써도 될 것 같아요 ! 고생하셨습니다 :)

ShinJongUng committed:
✨ feat: use FastImage library & change styles (#12)

props로 imgUrl을 받아주는 방식으로 제작, FastImage에는 Stylesheet 적용

#12
